scipy/optimize/nnls/nnls.f

```diff
@@ -197,7 +199,7 @@ SUBROUTINE NNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,MODE,MAXITER)
       ITER=ITER+1   
       IF (ITER .gt. ITMAX) then
          MODE=3
-         write (*,'(/a)') ' NNLS quitting on iteration count.'
+         ITERATIONQUIT=1
```

Comment:
Is there really a need for a new flag? You could simply remove the `write` command, and check for `mode==3` in the calling in the `nnls.py` wrapper routine:
```python
   if mode != 1:
        if mode == 3:
            logging.info("NNLS quitting on iteration count.")
        raise RuntimeError("too many iterations")
```